src/org/exist/xmldb/LocalCollection.java

```diff
@@ -249,7 +249,7 @@ public int getChildCollectionCount() throws XMLDBException {
 
     @Override
     public String getName() throws XMLDBException {
-        return this.<String>read().apply((collection, broker, transaction) -> collection.getURI().toString());
```

Comment:
I can't find records after 4 months.

Do you agree that you mixing up two semantically different methods into one? There are no way to prepare "good" error message for LocalCollection that do not reference to non existing collection because no what to get that collection url. That is bad design, IMHO.